networking_nsxv3/common/synchronization.py

```diff
@@ -187,9 +184,19 @@ def start(self):
 
     def stop(self):
         """ Gracefully terminates the runner instance """
+        while True:
```

Comment:
That doesn't look right to me: You are polling the `active` and `passive` queue, but do not seem to set any flag/condition to avoid new work being put into the queues.
Is that done externally? I would like that the termination logic is self-contained in the `stop()` routine.

Reply:
The shutdown logic is applicable only to the case when the agent is run as CLI with stopped synchronization. Then I am expecting termination after all of the work have been completed. I left timeout termination to external system, as I cannot determine when is the best time to kill an active synchronization process.

Reply:
I think, I didn't manage to get my point across here. It is essentially the same as: https://github.com/sapcc/networking-nsx-t/pull/62#discussion_r626828874

Reply:
- [ ] Prevent Runner to take new job when not in state "started"